includes/Data/Patterns.php

```diff
@@ -304,10 +289,54 @@ public static function get_theme_step_patterns_from_step( $step, $squash = false
 				}
 			}
 		}
-
 		return $squash ? $block_patterns_squashed : $block_patterns;
 	}
 
+	/**
+	 * Retrieve Homepage Menu Step Patterns
+	 *
+	 * @return array
+	 */
+	public static function get_patterns_for_homepage_menu_slugs() {
+		// fetch the selected header menu slug from DB
+		$flow_data        = \get_option( Options::get_option_name( 'flow' ) );
+		$header_menu_slug = explode( '/', $flow_data['data']['partHeader'] )[1];
+		$header_slug      = ! empty( $header_menu_slug ) ? $header_menu_slug : 'site-header-left-logo-navigation-inline';
```

Comment:
Don't we have a function named `replace_header_menu_slug` that can handle this for us?
``` 
$flow_data  = \get_option( Options::get_option_name( 'flow' ) );
$header_menu_slug = explode( '/', $flow_data['data']['partHeader'] )[1];
``` 
Also, can we check if we can move duplicate logic from `get_theme_step_patterns_from_step` and this one into that function